### ISSUE TITLE
SequelizeDatabaseError when ordering eager loaded associations with limit

### ISSUE THREAD

**Issue by bestan** [sequelize/sequelize#4762]
Getting `Unhandled rejection SequelizeDatabaseError: missing FROM-clause entry for table "children"` error when ordering eager loaded associations (`3.13.0`, postgres).

Data structure 
```javascript
var Parent = sequelize.define('parent');
var Child = sequelize.define('child');

Parent.hasMany(Child);
Child.belongsTo(Parent);
```

Code
```javascript
Parent.findAll({
  include: [{
    model: Child,
    limit: 5
  }],
  order: [ [Child, 'id'] ]
}).then(function(parents) {
});
```

It seems that `limit: 5` splits the query into two separate ones - select parents and then select children. Ordering by children is then performed within the **first** query (select parents), hence the `missing FROM-clause entry for table "children"`.

Produced SQL statement (where it crashes)
```sql
SELECT "parent"."id", "parent"."created_at", "parent"."updated_at" FROM "parents" AS "parent" ORDER BY "children"."id" LIMIT 5;
```

Second SQL statement (can be reached only after ordering is removed)
```sql
SELECT "child".* FROM ((SELECT "id", "created_at", "updated_at", "parent_id" FROM "children" AS "child" WHERE "child"."parent_id" = 1 LIMIT 5)) AS "child";
```

Full error stack trace
```
Unhandled rejection SequelizeDatabaseError: missing FROM-clause entry for table "children"
    at Query.formatError (/project/node_modules/sequelize/lib/dialects/postgres/query.js:443:14)
    at null.<anonymous> (/project/node_modules/sequelize/lib/dialects/postgres/query.js:118:19)
    at emitOne (events.js:77:13)
    at emit (events.js:169:7)
    at Query.handleError (/project/node_modules/pg/lib/query.js:108:8)
    at null.<anonymous> (/project/node_modules/pg/lib/client.js:171:26)
    at emitOne (events.js:77:13)
    at emit (events.js:169:7)
    at Socket.<anonymous> (/project/node_modules/pg/lib/connection.js:109:12)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)
```

**Comment by bestan** [comment]
Solved my use case with a raw query containing a [window function](http://www.postgresql.org/docs/9.4/static/tutorial-window.html) that looks something like this:

```sql
WITH children_ordered AS (
    SELECT *, row_number() OVER (
      	PARTITION BY parent_id ORDER BY id
    ) as row_number
    FROM children
)

SELECT * 
FROM parents
INNER JOIN children_ordered as children 
	ON children.parent_id = parents.id
WHERE row_number <= 5
```

Not sure if that's something Sequelize should be doing out of the box though.

**Comment by janmeier** [comment]
@bestan Glad you solved it! :) Unfortunately window functions are not supported by all dialects, so we can't use it as a general solution.

Did you try placing the `order` on the include? Since you are ordering the included and not the parent rows?

**Comment by bestan** [comment]
@janmeier oh well, thanks for pointing it out! 

I did try that earlier and it didn't work - children were not ordered as if order clause wasn't there, so I assumed that order can't be placed within include. However, I've tried it again just now and it works fine. I guess I've misspelled something yesterday.

An example with include in http://docs.sequelizejs.com/en/latest/docs/querying/#ordering would be helpful - I wouldn't make wrong assumptions.
